### PR TITLE
Remove unnecessary columns in the dataset processor

### PR DESCRIPTION
**Pull request by viswavi**
# Description

Earlier the dataset processor took the dataset given and added two new columns, `model_input` and `model_output`. This could be problematic, if the original dataset contained columns that could not be concatenated with the generated data (e.g. numerical values).

This PR removes these columns, keeping only the two necessary.

# References

N/A

# Blocked by

N/A

### CODE REVIEW

**Comment by zhaochenyang20** [comment]
> This could be problematic if the original dataset contained columns that could not be concatenated with the generated data (e.g. numerical values).

This happened when I was testing cli_demo. I merely convert all of it to string. Removing unnecessary columns is appropriate, but the unit tests for dataset processor should also be changed.

**Comment by zhaochenyang20** [comment]
It's weird that in my unit tests of `DatasetProcessor`, I assert there are four columns in the processed datasets. But this branch only returns four columns. I don't know why it passed the unit test. Possibly, previous unit tests were wrong. 😅

**Comment by zhaochenyang20** [comment]
```
[DatasetDict({
     train: Dataset({
         features: ['input_col', 'output_col', 'model_input', 'model_output'],
         num_rows: 2
     })
     test: Dataset({
         features: ['input_col', 'output_col', 'model_input', 'model_output'],
         num_rows: 2
     })
 }),
 DatasetDict({
     train: Dataset({
         features: ['input_col', 'output_col', 'model_input', 'model_output'],
         num_rows: 2
     })
     val: Dataset({
         features: ['input_col', 'output_col', 'model_input', 'model_output'],
         num_rows: 2
     })
 })]
 ```
 
 You see that these columns still exist. 🤔 @viswavi v

File: tests/dataset_processor_test.py [hunk 1]
```diff
@@ -188,6 +180,88 @@ def test_dataset_processor_t5_style():
     gc.collect()
 
 
+def test_dataset_processor_non_numerical_column():
+    """Test the `process_dataset_dict` function of T5-type `TextualizeProcessor`."""
```
Comment (zhaochenyang20) [inline tests/dataset_processor_test.py:184]: We should test a numerical column, right? 🤔

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]

File: tests/dataset_processor_test.py (hunk 1 above)
Comment (viswavi) [reply tests/dataset_processor_test.py]: Oh right 🤦  fixed the comment

**Review by zhaochenyang20 — APPROVED** [review]
cool! Wait for the CI and merge it!

**Pull request merged**